station-recipes-v0/src/main/java/net/modificationstation/stationapi/api/recipe/FuelRegistry.java

```diff
@@ -33,15 +56,31 @@ public static void addFuelTag(TagKey<Item> tag, int fuelTime) {
     @API
     public static void addFuelItem(Item item, int fuelTime) {
         viewInvalidated = true;
-        ITEM_FUEL_TIME.put(item, fuelTime);
+        ITEM_FUEL_TIME.computeIfAbsent(item, (o) -> new Int2IntOpenHashMap()).put(-1, fuelTime);
+    }
+
+    @API
+    public static void addFuelItem(ItemStack itemStack, int fuelTime) {
```

Comment:
This should probably make it clear that only metadata is taken from the item stack, since there's also count and custom NBT. Maybe replace ItemStack with Item and metadata int, or clarify in the function's name that it's for metadata only.